aws/resource_aws_fsx_lustre_file_system.go

```diff
@@ -154,6 +161,17 @@ func resourceAwsFsxLustreFileSystemCreate(d *schema.ResourceData, meta interface
 		},
 	}
 
+	var t string
+	if v, ok := d.GetOk("deployment_type"); ok {
+		t = v.(string)
+	}
+
+	if v, ok := d.GetOk("kms_key_id"); ok {
+		if t == fsx.LustreDeploymentTypePersistent1 {
```

Comment:
let remove the type check, AFAIK this kind of check isn't usually done in terraform as this should return an error to the user as the actual API would for an illegal combination.

Reply:
we can add a comment that this is relevant only for `TypePersistent1` instead of the required one

Reply:
removed